### ISSUE TITLE
[Retail Bug Report] "SetScale must be > 0" lua errors with black boxes

### ISSUE THREAD

**Issue by santa6666** [tukui-org/ElvUI#663]
### What is the issue you are having?

After updating from 13.08 to 13.09 I get tons of different lua errors about "SetScale has to be > 0"
Also getting auctionhouse errors.
On login it creates two black boxes that vanishes after a while. The lua errors shows on any load such as entering instance, typing /reload etc.
It also lowers fps.
Others in discord are reporting the identical issue.

### What is the expected behavior?

Not have these lua errors and drop in fps.

### What actually happened?

No idea. I updated and got the issues.
Others in discord are reporting the identical issue.

### Suggested Solution/Workaround

No idea. I'm guessing its something in the code as others are reporting my identical lua errors after updating to 13.09.

### Errors

```shell
Here is the /edebug discord message
https://discord.com/channels/209244641537556480/222356141785612288/1048749073773113406
Screenshot:
https://cdn.discordapp.com/attachments/222356141785612288/1048749073773105183/image.png

Lua errors:
20x MultiBarRight:SetScale(): Scale must be > 0
[string "=[C]"]: in function `SetScaleBase'
[string "@FrameXML/EditModeSystemTemplates.lua"]:48: in function `SetScale'
[string "@FrameXML/EditModeManager.lua"]:584: in function `UpdateRightActionBarPositions'
[string "@FrameXML/EditModeManager.lua"]:556: in function `UpdateActionBarLayout'
[string "@FrameXML/EditModeSystemTemplates.lua"]:676: in function `UpdateGridLayout'
[string "@FrameXML/ActionBar.lua"]:350: in function `UpdateVisibility'
[string "@FrameXML/ActionBar.lua"]:317: in function `HideOverride'
[string "@FrameXML/ActionBar.lua"]:302: in function `SetShown'
[string "@FrameXML/MultiActionBars.lua"]:35: in function <FrameXML/MultiActionBars.lua:30>
[string "@FrameXML/MultiActionBars.lua"]:43: in function `MultiActionBar_Update'
[string "@FrameXML/UIParent.lua"]:1672: in function <FrameXML/UIParent.lua:1362>

Locals:
(*temporary) = MultiBarRight {
 0 = <userdata>
 buttonPadding = 2
 UpdateSystemSettingAlwaysShowButtons = <function> defined @FrameXML/EditModeSystemTemplates.lua:793
 UpdateMagnetismRegistration = <function> defined @FrameXML/EditModeSystemTemplates.lua:527
 GetSnapOffsets = <function> defined @FrameXML/EditModeSystemTemplates.lua:421
 UpdateFrameStrata = <function> defined @FrameXML/ActionBar.lua:194
 UpdateButtonArt = <function> defined @FrameXML/EditModeSystemTemplates.lua:690
 IsFrameAnchoredToMe = <function> defined @FrameXML/EditModeSystemTemplates.lua:490
 buttonTemplate = "MultiBar3ButtonTemplate"
 ShowBase = <function> defined =[C]:-1
 IsShown = <function> defined @FrameXML/ActionBar.lua:286
 GetSelectionOffset = <function> defined @FrameXML/EditModeSystemTemplates.lua:373
 ResetToDefaultPosition = <function> defined @FrameXML/EditModeSystemTemplates.lua:158
 RefreshButtonArt = <function> defined @FrameXML/EditModeSystemTemplates.lua:683
 GetScaledSelectionSides = <function> defined @FrameXML/EditModeSystemTemplates.lua:365
 SelectSystem = <function> defined @FrameXML/EditModeSystemTemplates.lua:555
 buttonArtDirty = false
 GetFrameMagneticEligibility = <function> defined @FrameXML/EditModeSystemTemplates.lua:510
 AddLayoutChildren = <function> defined @SharedXML/LayoutFrame.lua:48
 IsBelowFrame = <function> defined @FrameXML/EditModeSystemTemplates.lua:332
 UpdateSystemSettingIconSize = <function> defined @FrameXML/EditModeSystemTemplates.lua:735
 SetScaleBase = <function> defined =[C]:-1
 HasActiveChanges = <function> defined @FrameXML/EditModeSystemTemplates.lua:254
 GetScaledCenter = <function> defined @FrameXML/EditModeSystemTemplates.lua:358
 DoesSettingDisplayValueEqual = <function> defined @FrameXML/EditModeSystemTemplates.lua:278
 Layout = <function> defined @SharedXML/LayoutFrame.lua:333
 settingDisplayInfoMap = <table> {
 }
 GetSettingValueBool = <function> defined @FrameXML/EditModeSystemTemplates.lua:269
 IsLayoutFrame = <function> defined @SharedXML/LayoutFrame.lua:36
 DoesSettingValueEqual = <function> defined @FrameXML/EditModeSystemTemplates.lua:273
 MarkDirty = <function> defined @SharedXML/LayoutFrame.lua:90
 ShouldResetSettingsDialogAnchors = <function> defined @FrameXML/EditModeSystemTemplates.lua:101
 dirtySettings = <table> {
 }
 UpdateGridLayout = <function> defined @FrameXML/EditModeSystemTemplates.lua:669
 IsToTheLeftOfFrame = <function> defined @FrameXML/EditModeSystemTemplates.lua:314
 GetSettingsDialogAnchor = <function> defined @FrameXML/EditModeSystemTemplates.lua:298
 hasActiveChanges = false
 OnSystemLoad = <function> defined @FrameXML/EditModeSystemTemplates.lua:3
 ClearFrameSnap = <function> defined @FrameXML/EditModeSystemTemplates.lua:458
 GetCombinedCenterOffset = <function> defined @FrameXML/EditModeSystemTemplates.lua:408
 SetupSettingsDialogAnchor = <function> defined @FrameXML/EditModeSystemTemplates.lua:242
 IsShownOverride = <function> defined @FrameXML/ActionBar.lua:286
 dirty = false
 SetPointBase = <function> defined =[C]:-1
 UpdateSystem = <function> defined @FrameXML/EditModeSystemTemplates.lua:607
 UpdateShownButtons = <function> defined @FrameXML/ActionBar.lua:198
 UpdateDirtySettings = <function> defined @FrameXML/EditModeSystemTemplates.lua:122
 BreakSnappedFrames = <function> defined @FrameXML/EditModeSystemTemplates.lua:441
 HideOverride = <function> defined @FrameXML/ActionBar.lua:313
 PrepareForSave = <function> defined @FrameXML/EditModeSystemTemplates.lua:35
 systemIndex = 4
 snappedFrames = <table> {
 }
 SetShownOverride = <function> defined @FrameXML/ActionBar.lua:298
 SetPointOverride = <function> defined @FrameXML/EditModeSystemTemplates.lua:71
 
Another lua error:
1x SharedXML/Scroll/ScrollBoxListView.lua:130: attempt to index a nil value
[string "@SharedXML/Scroll/ScrollBoxListView.lua"]:130: in function <SharedXML/Scroll/ScrollBoxListView.lua:129>
[string "=(tail call)"]: ?
[string "@SharedXML/Scroll/ScrollBox.lua"]:680: in function `ScrollToElementDataIndex'
[string "@Blizzard_AuctionHouseUI/Blizzard_AuctionHouseCommoditiesList.lua"]:163: in function `SetQuantitySelected'
[string "@Blizzard_AuctionHouseUI/Blizzard_AuctionHouseCommoditiesList.lua"]:182: in function <...tionHouseUI/Blizzard_AuctionHouseCommoditiesList.lua:181>
[string "=[C]"]: ?
[string "@SharedXML/CallbackRegistry.lua"]:143: in function <SharedXML/CallbackRegistry.lua:142>
[string "=[C]"]: ?
[string "@SharedXML/CallbackRegistry.lua"]:146: in function `TriggerEvent'
[string "@Blizzard_AuctionHouseUI/Blizzard_AuctionHouseCommoditiesBuyFrame.lua"]:30: in function `inputChangedCallback'
[string "@Blizzard_AuctionHouseUI/Blizzard_AuctionHouseSharedTemplates.lua"]:558: in function <...tionHouseUI/Blizzard_AuctionHouseSharedTemplates.lua:552>

Locals:
self = <table> {
 EnumerateFrames = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:113
 GetFrameExtent = <function> defined @SharedXML/Scroll/ScrollController.lua:11
 Flush = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:93
 SetAcquireLocked = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:232
 GetDataScrollOffset = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:701
 SetDataProvider = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:192
 ForEachFrame = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:107
 Rebuild = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:707
 InsertElementDataTableRange = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:145
 dataIndexEnd = 0
 SelectPointComponent = <function> defined @SharedXML/Scroll/ScrollController.lua:37
 dataIndexBegin = 0
 extent = 0
 SetFrameExtent = <function> defined @SharedXML/Scroll/ScrollController.lua:16
 CalculateFrameExtent = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:397
 GetIdenticalElementExtents = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:628
 GetDataIndexEnd = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:816
 GetUpper = <function> defined @SharedXML/Scroll/ScrollController.lua:24
 ClearDataProvider = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:179
 padding = <table> {
 }
 ClearDataProviderInternal = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:165
 templateInfos = <table> {
 }
 SetHorizontal = <function> defined @SharedXML/Scroll/ScrollController.lua:3
 initializerList = <table> {
 }
 poolCollection = <table> {
 }
 frames = <table> {
 }
 SignalDataChangeEvent = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:223
 callbackTables = <table> {
 }
 scrollBox = Frame {
 }
 GetElementExtentCalculator = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:675
 ClearInvalidation = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:804
 CalculateDataIndices = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:109
 GetScrollTarget = <function> defined @SharedXML/Scroll/ScrollBoxView.lua:54
 FindElementDataIndexByPredicate = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:121
 HasDataProvider = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:157
 GetInvalidationReason = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:800
 GetCallbacksByEvent = <function> defined @SharedXML/CallbackRegistry.lua:54
 IsAcquireLocked = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:228
 UnregisterEvents = <function> defined @SharedXML/CallbackRegistry.lua:165
 UnassignAccessors = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:86
 GetFrameCount = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:321
 SetElementExtentCalculator = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:668
 GetTemplateFromElementData = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:417
 GetCallbackTable = <function> defined @SharedXML/CallbackRegistry.lua:50
 GetFrameLevelPolicy = <function> defined @SharedXML/Scroll/ScrollBoxView.lua:9
 GetPanExtent = <function> defined @SharedXML/Scroll/ScrollBoxLinearView.lua:125
 elementFactory = <function> defined @Blizzard_AuctionHouseUI/Blizzard_AuctionHouseItemList.lua:141
 HasEqualTemplateInfoExtents = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:606
 SetFrameLevelPolicy = <function> defined @SharedXML/Scroll/ScrollBoxView.lua:13
 CreateTemplateExtentFromElementData = <function> defined @SharedXML/Scroll/ScrollBoxListView.lua:430
 Layout = <function> defined @SharedXML/
```

### ElvUI Changelog & Troubleshooting

- [X] I have confirmed that I have read the [Changelog](https://www.tukui.org/download.php?ui=elvui&changelog).
- [X] I have verfied ElvUI is up to date by checking /estatus.
- [X] I have verfied my issue doesn't exist in the ElvUI Issue Tracker.
- [ ] I have not read step six and I am just clicking all boxes.
- [X] I have verfied my issue doesn't exist in the default UI.
- [X] I have disabled all addons except the three called 'ElvUI', 'ElvUI Libraries', and 'ElvUI Options' by using `/edebug on`? This is important in order to rule out that any of your other addons are causing the issue. If you manage to pinpoint a conflict between ElvUI and another addon, make sure you include that information.

**Comment by santa6666** [comment]
On repocc suggestion I tried the following:
Went into interface and manually deleted the 3 elvui folders.
Downloaded zip from webpage and manually installed them again.
Also went into WTF folder and searched for "edit-mode" and deleted all those files. Both in account and under character folder.
Error still persists through both fixes.

**Comment by santa6666** [comment]
So its easy to find:
Here is the /edebug discord message
https://discord.com/channels/209244641537556480/222356141785612288/1048749073773113406
Screenshot:
https://cdn.discordapp.com/attachments/222356141785612288/1048749073773105183/image.png

**Comment by kodewdle** [comment]
fixed in 13.10, thanks! 🌼
